spec/unit/resource_spec.rb

```diff
@@ -250,7 +254,12 @@ def config(options = {})
         let(:different_post) { double }
         before do
           allow(Post).to receive(:primary_key).and_return 'something_else'
-          allow(Post).to receive(:find_by_something_else).with('55555') { different_post }
+          if Rails::VERSION::MAJOR > 3
+            allow(Post).to receive(:find_by).
+              with("something_else" => "55555") { different_post }
+          else
+            allow(Post).to receive(:find_by_something_else).with('55555') { different_post }
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.